### PR TITLE
Spellbook workshop submission

### PR DESCRIPTION
**Pull request by bernatfp**
Brief comments on the purpose of your changes:

*For Dune Engine V2*
I've checked that:
General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if adding a new model, I edited the alter table macro to display new database object (table or view) in UI explorer
* [ ] if adding a new materialized table, I edited the optimize table macro

Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by bernatfp** [comment]
Don't merge, this is just demonstrative. Wanted to try it out end to end.

File: models/balancer/balancer_ethereum_balances.sql [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        SELECT p.pools as pool, date_trunc('day', e.evt_block_time) AS day, e.contract_address AS token, SUM(value) AS amount
+        FROM {{ source('erc20_ethereum', 'evt_transfer') }} e 
+        INNER JOIN pools p ON e.`to` = p.pools
+        GROUP BY 1, 2, 3
+
+        UNION ALL
+
+        SELECT e.`to` as pool, date_trunc('day', e.evt_block_time) AS day, e.contract_address AS token, SUM(value) AS amount
+        FROM {{ source('erc20_ethereum', 'evt_transfer') }} e
+        WHERE e.`to` = '0xBA12222222228d8Ba445958a75a0704d566BF2C8'
```
Comment (masquot) [inline models/balancer/balancer_ethereum_balances.sql:17]: Here and below as well, you need a lower-case address. Simple way to convert:
https://dune.com/queries/1274287

File: macros/alter_table_properties.sql [hunk 2]
```diff
@@ -685,6 +685,15 @@ ALTER VIEW archipelago_ethereum.fees  SET TBLPROPERTIES ('dune.public'='true',
                                                         'dune.data_explorer.contributors'='["0xRob"]');
 {% endset %}
 
+{% set balancer_ethereum_balances %}
+ALTER VIEW balancer_ethereum.balances SET TBLPROPERTIES('dune.public'='true',
+                                                   'dune.data_explorer.blockchains'='["ethereum"]',
+                                                   'dune.data_explorer.category'='abstraction',
+                                                   'dune.data_explorer.abstraction.type'='project',
+                                                   'dune.data_explorer.abstraction.name'=balancer,
+                                                   'dune.data_explorer.contributors'='["you"]');
```
Comment (masquot) [inline macros/alter_table_properties.sql:694]: update contributor field to your user name

**Review by masquot — CHANGES_REQUESTED** [review]

**Comment by dot2dotseurat** [comment]
Thank you for participating in our Spellbook workshop competition! It was a close race and although your submission didn't win, there will be future opportunities to participate in Spellbook hackathons and bounties (launching soon). we hope you learned something new and had some fun. and that you will support team abstract in building an in browser IDE.

**Pull request closed without merging**